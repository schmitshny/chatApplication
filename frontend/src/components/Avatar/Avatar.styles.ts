import styled, { css } from 'styled-components';
import { AvatarSize } from './Avatar';

export type AvatarShape = 'circle' | 'rounded';

interface StyledAvatarProps {
  size: AvatarSize;
  shape?: AvatarShape;
}

interface StatusDotProps {
  $isOnline: boolean;
}

export const AvatarContainer = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const StyledAvatar = styled.img<StyledAvatarProps>`
  border-radius: ${({ shape }) => (shape === 'circle' ? '50%' : '16px')};
  border: 1px solid silver;
  align-self: center;

  ${(props) => {
    switch (props.size) {
      case 'small':
        return css`
          width: 38px;
          height: 38px;
        `;
      case 'medium':
        return css`
          width: 50px;
          height: 50px;
        `;
      case 'large':
        return css`
          width: 70px;
          height: 70px;
        `;
      default:
        return css`
          width: 50px;
          height: 50px;
        `;
    }
  }}
`;

export const StatusDot = styled.span<StatusDotProps>`
  position: absolute;
  bottom: 0px;
  right: 0;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  border: 1px solid black;
  background-color: ${({ $isOnline }) => ($isOnline ? 'green' : 'gray')};
`;
