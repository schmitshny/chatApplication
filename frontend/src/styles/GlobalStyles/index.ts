import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  body {
    background-color: ${({ theme }) => theme.colors.background};
    font-family: ${({ theme }) => theme.fonts.primary};
    color: ${({ theme }) => theme.colors.text};
  }
`;
