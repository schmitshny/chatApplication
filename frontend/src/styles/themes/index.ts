export const lightTheme = {
  colors: {
    background: '#EEE',
    secondaryBackground: '#FFF',
    contrastBackground: '#393939',
    activeBackground: '#F5F5F5',
    hoverBackground: '#F5F5D1',
    text: '#000',
    contrastText: '#FFF',
    buttonText: '#000',
    sectionHeader: '#5E5B5B',
    iconBackground: '#fff',
    linkColor: '#000',
    inputText: '#222',
    info: '#3498db',
    error: '#e74c3c',
    success: '#2ecc71',
  },
  fonts: {
    primary: "'Poppins', sans-serif",
  },
  shadows: {
    primary: '0 15px 16.83px 0.17px rgba(0, 0, 0, 0.05)',
    avatarShadow: 'rgba(0, 0, 0, 0.2)',
  },
};

export const darkTheme = {
  colors: {
    background: '#262626',
    secondaryBackground: '#1e1e1e',
    contrastBackground: '#eee',
    activeBackground: '#393939',
    hoverBackground: '#292829',
    text: '#fff',
    contrastText: '#000',
    buttonText: '#fff',
    sectionHeader: '#B7B7B7',
    iconBackground: '#2E2E2E',
    inputText: '#eee',
    info: '#5DADE2',
    error: '#e72c5c',
    success: '#58D68D',
  },
  fonts: {
    primary: "'Poppins', sans-serif",
  },
  shadows: {
    primary: '0 11px 12.83px 0.17px rgba(0, 0, 0, 0.1)',
    avatarShadow: 'rgba(255, 255, 255, 0.2)',
  },
};
