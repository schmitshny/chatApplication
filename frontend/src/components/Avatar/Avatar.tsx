import { AvatarContainer, AvatarShape, StatusDot, StyledAvatar } from './Avatar.styles';
import defaultAvatar from './assets/defaultAvatar.jpg';

export type AvatarSize = 'small' | 'medium' | 'large';

interface AvatarProps {
  avatarUrl?: string;
  size?: AvatarSize;
  shape?: AvatarShape;
  isOnline?: boolean;
  displayStatus?: boolean;
}

export const Avatar = ({
  avatarUrl,
  size = 'medium',
  isOnline = false,
  displayStatus,
  shape = 'circle',
}: AvatarProps) => {
  return (
    <AvatarContainer>
      <StyledAvatar src={avatarUrl || defaultAvatar} alt="user avatar" size={size} shape={shape} loading="lazy" />
      {displayStatus && <StatusDot $isOnline={isOnline} data-testid="status-dot" />}
    </AvatarContainer>
  );
};
