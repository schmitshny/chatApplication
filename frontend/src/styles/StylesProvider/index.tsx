import { ThemeProvider } from '../../features/theme/ThemeProvider';
import { GlobalStyles } from '../GlobalStyles';

interface StylesProviderProps {
  children: React.ReactNode;
}

export const StylesProvider = ({ children }: StylesProviderProps) => (
  <ThemeProvider>
    <GlobalStyles />
    {children}
  </ThemeProvider>
);
