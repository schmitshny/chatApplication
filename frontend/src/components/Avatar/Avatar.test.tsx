import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import { Avatar } from '.';
import defaultAvatar from './assets/defaultAvatar.jpg';
import { AvatarShape } from './Avatar.styles';

describe('Avatar component', () => {
  test('should render correctly', () => {
    const { container } = render(<Avatar />);

    const avatarImage = screen.getByAltText('user avatar');
    expect(avatarImage).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  test('displays custom avatar image', () => {
    const { container } = render(<Avatar avatarUrl="https://example.com/avatar.png" />);

    const avatarImage = screen.getByRole('img', { name: 'user avatar' });
    expect(avatarImage).toBeInTheDocument();
    expect(avatarImage).toHaveAttribute('src', 'https://example.com/avatar.png');

    expect(container).toMatchSnapshot();
  });

  test.each<AvatarShape>(['circle', 'rounded'])('displays with shape %s', (shape) => {
    const { container } = render(<Avatar shape={shape} />);

    expect(container).toMatchSnapshot();
  });

  test.each([true, false])('displays status dot when displayStatus is %s', (displayStatus) => {
    render(<Avatar displayStatus={displayStatus} />);

    const statusDot = screen.queryByTestId('status-dot');

    displayStatus ? expect(statusDot).toBeInTheDocument() : expect(statusDot).not.toBeInTheDocument();
  });

  test('displays default avatar image when no URL is provided', () => {
    render(<Avatar />);
    const avatarImage = screen.getByRole('img', { name: 'user avatar' });

    expect(avatarImage).toHaveAttribute('src', defaultAvatar);
  });
});
