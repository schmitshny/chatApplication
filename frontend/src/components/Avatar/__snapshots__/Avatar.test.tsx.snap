// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > displays custom avatar image 1`] = `
<div>
  <div
    class="sc-bczSft iUPaN"
  >
    <img
      alt="user avatar"
      class="sc-gsnScm eXjLEN"
      loading="lazy"
      shape="circle"
      src="https://example.com/avatar.png"
    />
  </div>
</div>
`;

exports[`Avatar component > displays with shape circle 1`] = `
<div>
  <div
    class="sc-bczSft iUPaN"
  >
    <img
      alt="user avatar"
      class="sc-gsnScm eXjLEN"
      loading="lazy"
      shape="circle"
      src="/src/components/Avatar/assets/defaultAvatar.jpg"
    />
  </div>
</div>
`;

exports[`Avatar component > displays with shape rounded 1`] = `
<div>
  <div
    class="sc-bczSft iUPaN"
  >
    <img
      alt="user avatar"
      class="sc-gsnScm dTwvaW"
      loading="lazy"
      shape="rounded"
      src="/src/components/Avatar/assets/defaultAvatar.jpg"
    />
  </div>
</div>
`;

exports[`Avatar component > should render correctly 1`] = `
<div>
  <div
    class="sc-bczSft iUPaN"
  >
    <img
      alt="user avatar"
      class="sc-gsnScm eXjLEN"
      loading="lazy"
      shape="circle"
      src="/src/components/Avatar/assets/defaultAvatar.jpg"
    />
  </div>
</div>
`;
